pkg/controller/ec2/vpc/controller.go

```diff
@@ -191,17 +192,22 @@ func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.Ex
 		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
 	}
 
-	for _, input := range []*awsec2.ModifyVpcAttributeInput{
-		{
+	if cr.Spec.ForProvider.EnableDNSSupport != nil {
```

Comment:
I don't love that we will make _all_ of these calls if the fields are set even if they don't need to be changed (i.e. the diff was elsewhere in the spec). That being said, this is the current existing pattern (as opposed to getting again, calculating diff, then only making the specific call), so I am fine with moving forward with this fix 👍

Reply:
@hasheddan I think I just ran into a bug here. So in my case I'm importing and existing vpc. I'm using the `crossplane.io/external-name` annotation in order to pull in an existing vpc. In doing so I'm getting an error of:

```
 Message:               update failed: failed to modify the VPC resource attributes: api error UnauthorizedOperation: You are not authorized to perform this operation. Encoded authorization failure message: 17IZjLqa5Zh3OjVaZ3t9FH0FftiP1SQ3_yRUhVUg42S9N-S5HC6P5UNgad1lXNzm9uEfvQyWMV-JzvjbCbdBoeCDdaAeadxZWWO5F94c-Qm0ba2sMINryty1b1F_wuzxG9Pi_1Mb9QEwuMSjR2zK7Bttkm20YGfP-NKxSIenBSHVCmy5yfW915cTLqakozjE8Dmt2Ml95_XT8nQ5QZj-bYHkCIu5WjTY5VQQNe7Ur7hPXNoj2Zqp3mUnwpJKmPp0XAHgRrQtnClrc_aFldtspgv2Ty97AhgIQtOWHIvX4iadQ1uafOakWfKoONODQaUw8zqHHsBNt6a1-Cy7nQ9Q1weM0tyeVykTioT1iRqq2arbvdvSjWvPq9P8qqpkovXFBqxRExx8wibFiZ6G5bV3lmAv4wMZOa11p0nuisNmnc2xJXDb8gyX2dwSQSXL336pRHgKt4S-VITej52_-o-qD7wqEzmN6rZibbfIA2xwGddtEBX95zwgqJcgWLIr-Rvz4q4ktgLuIJH5G5rOkmH1ZiCMTVswY98WjouXAl9bXUaNnp55LzvhSdNajsW77Uuu5kDlc0A4WAUrdaGD6He8foMacrySqBwyrlJLULa-NxkHcKC5p2tt5i3k-BYurVJswQnqXTIbmH4lwV5GIimXuPkcaGX9lKXhvGUl4zAiYCSmG_QweVQT0cglv8dg9R4oOwws8WNtDM_aHQbDjxO13SO1I17geD1ERFiHqdN7igqbPLs2aRUA7f5vVS7Z-DsF-p5Mf5uke7qv-4eyd7on2bV-e-rDrH5W6KfwoyBLvbq7xZGY7rIG3is9aQgDoBFwk2NVcIAKPlP0OQ2y0OKxx83q_mUcDcpsYdz5ZmZjghiaJr9UMxzEfnuEwTvBwVPqRJiw1CdWhoqmMoIPGg
```

In checking CloudTrail I see it's attempting to do a `ModifyVpcResource` on `enableDnsSupport` even though it's already set. 

Obviously in our case we are using vpc as a ReadOnly resource for reference but it shouldn't try to Modify an attribute if it matches. I've tried setting the forProvider to not set these 2 attributes and it still tries to Modify... I've tried to set these forProvider parameters to match what they currently are on vpc and it still tries to modify.

Reply:
@hasheddan I think I just ran into a bug here. So in my case I'm importing and existing vpc. I'm using the `crossplane.io/external-name` annotation in order to pull in an existing vpc. In doing so I'm getting an error of:

```
 Message:               update failed: failed to modify the VPC resource attributes: api error UnauthorizedOperation: You are not authorized to perform this operation. Encoded authorization failure message: 17IZjLqa5Zh3OjVaZ3t9FH0FftiP1SQ3_yRUhVUg42S9N-S5HC6P5UNgad1lXNzm9uEfvQyWMV-JzvjbCbdBoeCDdaAeadxZWWO5F94c-Qm0ba2sMINryty1b1F_wuzxG9Pi_1Mb9QEwuMSjR2zK7Bttkm20YGfP-NKxSIenBSHVCmy5yfW915cTLqakozjE8Dmt2Ml95_XT8nQ5QZj-bYHkCIu5WjTY5VQQNe7Ur7hPXNoj2Zqp3mUnwpJKmPp0XAHgRrQtnClrc_aFldtspgv2Ty97AhgIQtOWHIvX4iadQ1uafOakWfKoONODQaUw8zqHHsBNt6a1-Cy7nQ9Q1weM0tyeVykTioT1iRqq2arbvdvSjWvPq9P8qqpkovXFBqxRExx8wibFiZ6G5bV3lmAv4wMZOa11p0nuisNmnc2xJXDb8gyX2dwSQSXL336pRHgKt4S-VITej52_-o-qD7wqEzmN6rZibbfIA2xwGddtEBX95zwgqJcgWLIr-Rvz4q4ktgLuIJH5G5rOkmH1ZiCMTVswY98WjouXAl9bXUaNnp55LzvhSdNajsW77Uuu5kDlc0A4WAUrdaGD6He8foMacrySqBwyrlJLULa-NxkHcKC5p2tt5i3k-BYurVJswQnqXTIbmH4lwV5GIimXuPkcaGX9lKXhvGUl4zAiYCSmG_QweVQT0cglv8dg9R4oOwws8WNtDM_aHQbDjxO13SO1I17geD1ERFiHqdN7igqbPLs2aRUA7f5vVS7Z-DsF-p5Mf5uke7qv-4eyd7on2bV-e-rDrH5W6KfwoyBLvbq7xZGY7rIG3is9aQgDoBFwk2NVcIAKPlP0OQ2y0OKxx83q_mUcDcpsYdz5ZmZjghiaJr9UMxzEfnuEwTvBwVPqRJiw1CdWhoqmMoIPGg
```

In checking CloudTrail I see it's attempting to do a `ModifyVpcResource` on `enableDnsSupport` even though it's already set. 

Obviously in our case we are using vpc as a ReadOnly resource for reference but it shouldn't try to Modify an attribute if it matches. I've tried setting the forProvider to not set these 2 attributes and it still tries to Modify... I've tried to set these forProvider parameters to match what they currently are on vpc and it still tries to modify.